### PR TITLE
[Fix](nereids) fix merge aggregate setting top projection bug

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
introduced by #31811
sql like this:
```sql
    select col1, col2 from  (select a as col1, a as col2 from mal_test1 group by a) t group by col1, col2 ;
```
Transformation Description:
In the process of optimizing the query, an agg-project-agg pattern is transformed into a project-agg pattern:
Before Transformation:

```
LogicalAggregate
+-- LogicalPrject
    +-- LogicalAggregate
```

After Transformation:

```
LogicalProject
+-- LogicalAggregate
```

Before the transformation, the projection in the LogicalProject was a AS col1, a AS col2, and the outer aggregate group by keys were col1, col2. After the transformation, the aggregate group by keys became a, a, and the projection remained a AS col1, a AS col2.

Problem:
When building the project projections, the group by key a, a needed to be transformed to a AS col1, a AS col2. The old code had a bug where it used the slot as the map key and the alias in the projections as the map value. This approach did not account for the situation where aliases might have the same slot.

Solution:
The new code fixes this issue by using the original outer aggregate group by expression's exprId. It searches within the original project projections to find the NamedExpression that has the same exprId. These expressions are then placed into the new projections. This method ensures that the correct aliases are maintained, resolving the bug.

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — COMMENTED** [review]
add test case.
add a description explaining the cause of the problem and the PR number that introduced it

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run p0

**Comment by feiniaofeiafei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java [hunk 1]
```diff
@@ -108,10 +113,18 @@ private Plan mergeAggProjectAgg(LogicalAggregate<LogicalProject<LogicalAggregate
...
+            exprIdToNameExpressionMap.put(pro.getExprId(), pro);
+        }
+        List<Expression> originOuterAggGroupBy = outerAgg.getGroupByExpressions();
+        List<Expression> projectGroupBy = new ArrayList<>();
+        for (int i = 0; i < originOuterAggGroupBy.size(); i++) {
+            ExprId exprId = ((NamedExpression) (originOuterAggGroupBy.get(i))).getExprId();
+            NamedExpression namedExpression = exprIdToNameExpressionMap.get(exprId);
+            projectGroupBy.add(namedExpression);
+        }
+        // List<Expression> projectGroupBy = ExpressionUtils.replace(replacedGroupBy, childToAlias);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergeAggregate.java:127]: remove useless code

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**